assets/js/googlesitekit/widgets/util/get-widget-layout.js

```diff
@@ -156,27 +141,36 @@ export function getWidgetClassNames( activeWidgets ) {
 		// Grab the width of the first size in the sizes list, it's either the default one or the best suiting to the current row.
 		const width = sizes[ 0 ].width;
 
+		// Populate initial column width and row index for the widget.
+		// These may be corrected below in certain situations.
+		columnWidths.push( WIDTH_GRID_COUNTER_MAP[ width ] );
+		rowIndexes.push( rowIndex );
+
 		// Increase column counter based on width.
 		counter += WIDTH_GRID_COUNTER_MAP[ width ];
 
-		// If counter is exactly 12, the next widget is going to be in a new row.
-		if ( counter % 12 === 0 ) {
-			counter = 0;
-		}
-
 		// If counter is going above 12, this widget is too wide for the current row.
 		// So it's going to be the first widget in the next row instead.
 		if ( counter > 12 ) {
 			counter -= WIDTH_GRID_COUNTER_MAP[ width ];
 
+			// Correct the previously added row index for this widget as it
+			// will end up in the following row.
+			rowIndexes[ i ]++;
+
 			// If the column count without the overflowing widget is exactly 9, expand
 			// the widths of these widgets slightly to fill the entire 12 columns.
 			if ( counter === 9 ) {
-				[ classNames, counter ] = resizeClasses( classNames, counter );
+				[ classNames, columnWidths, counter ] = resizeClasses( classNames, columnWidths, counter );
 			}
 
 			// See above, initial counter for the next row of widgets.
 			counter = WIDTH_GRID_COUNTER_MAP[ width ];
+			rowIndex++;
+		} else if ( counter % 12 === 0 ) {
```

Comment:
I think this can be simplified since the condition for greater than 12 is right before this.
```suggestion
		} else if ( counter === 12 ) {
```